test/pipeline-component-factory-mocha.js

```diff
@@ -0,0 +1,239 @@
+/**
+ * pipeline-component-factory-mocha.js
+ *
+ * This module tests the pipline component factory to verify that it creates a noflo component 
+ * with the appropriate RDF pipeline extensions.  Specifically RDF pipeline components should
+ * have the following node extensions: 
+ *   - A vni facade 
+ *   - A wrapper object to be used to hold a reference to the standard RDF framework 
+ *     wrapper functions: fRunUpdater, fExists, and fDestroyState. More details on wrappers
+ *     can be found here: https://github.com/rdf-pipeline/noflo-rdf-pipeline/wiki/Wrapper-API 
+ * In addition, rdf pipeline nodes should register an ondata callback function on each input port
+ * to be called whenever new data is received. 
+ */
+
+var chai = require('chai');
+var chaiAsPromised = require('chai-as-promised');
+chai.should();
+chai.use(chaiAsPromised);
+
+var _ = require('underscore');
+
+var pipelineCompFactory = require('../src/pipeline-component-factory.js');
+var stateFactory = require('../src/create-state');
+var test = require('./common-test');
+
+describe('pipeline-component-factory', function() {
+
+    it("should exist as an function", function() {
+        pipelineCompFactory.should.exist;
+        pipelineCompFactory.should.be.a('function');
+    });
+
+    it("should reject undefined node definition", function() {
+        return Promise.resolve().then(pipelineCompFactory).should.be.rejected;
+    });
+
+    it("should create an RDF pipeline component factory producing a node with pipeline extensions", function() {
+
+        // define a node
+        var nodeDef = {
+            description: "A component description",
+            inPorts: {
+               input: {
+                   datatype: 'string',
+                   description: "a port description",
+               }
+            }
+        }
+
+        // Get a pipeline factory and instantiate the component without passing in a wrapper parameter
```

Comment:
I am testing that it does not blow up if no wrapper is passed.  It will default to an empty object if no wrapper is defined.   Would you prefer an error?